plugins/inputs/http/http.go

```diff
@@ -108,31 +123,54 @@ func (*HTTP) Description() string {
 }
 
 func (h *HTTP) Init() error {
-	tlsCfg, err := h.ClientConfig.TLSConfig()
+	ctx := context.Background()
+	client, err := h.createClient(ctx)
 	if err != nil {
 		return err
 	}
 
+	h.client = client
+
+	// Set default as [200]
+	if len(h.SuccessStatusCodes) == 0 {
+		h.SuccessStatusCodes = []int{200}
+	}
+	return nil
+}
+
+func (h *HTTP) createClient(ctx context.Context) (*http.Client, error) {
+	tlsCfg, err := h.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
 	proxy, err := h.HTTPProxy.Proxy()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	transport := &http.Transport{
 		TLSClientConfig: tlsCfg,
 		Proxy:           proxy,
 	}
 
-	h.client = &http.Client{
+	client := &http.Client{
 		Transport: transport,
 		Timeout:   time.Duration(h.Timeout),
 	}
 
-	// Set default as [200]
-	if len(h.SuccessStatusCodes) == 0 {
-		h.SuccessStatusCodes = []int{200}
+	if h.ClientID != "" && h.ClientSecret != "" && h.TokenURL != "" {
+		oauthConfig := clientcredentials.Config{
+			ClientID:     h.ClientID,
+			ClientSecret: h.ClientSecret,
+			TokenURL:     h.TokenURL,
+			Scopes:       h.Scopes,
+		}
+		ctx = context.WithValue(ctx, oauth2.HTTPClient, client)
+		client = oauthConfig.Client(ctx)
```

Comment:
I might make sense to log (maybe as debug) if at least one of those fields is empty but at least one of the other fields isn't, i.e. if the user forgot to specify one of the fields.

Reply:
Agreed it would help, I'm not sure if it would be worth adding more complexity to the code though?

Reply:
From a user's perspective you might wonder why the connection fails if you forgot one of those and you will have no means to find out why. It might be just me, but spending extra lines to help the user finding problems in his/her config is worth the two lines
```golang
else if  h.ClientID != "" || h.ClientSecret != "" || h.TokenURL != "" {
    h.Log.Warnf("One of the fields client ID=%q, client secret=%q or token URL=%q is empty. Skipping OAuth.", h.ClientID, h.ClientSecret, h.TokenURL)
}
```

Reply:
Sure, I'll add that in. thanks!